README.md

```diff
@@ -894,35 +903,31 @@ template <class D, class P>
 class TorqueGeneratedStruct : public P {
  public:
 ```
-Where D is Struct and P is HeapObject in this case. But the above is the declartion
+Where D is Struct and P is HeapObject in this case. But the above is the declaration
 of the type but what we have in the .h file is what was generated. 
 
 This type is defined in `src/objects/struct.tq`:
 ```
-@abstract                                                                       
-@generatePrint                                                                  
-@generateCppClass                                                               
-extern class Struct extends HeapObject {                                        
-} 
+@abstract
+extern class Struct extends HeapObject {
+}
 ```
 
 `NewStruct` can be found in `src/heap/factory-base.cc`
 ```c++
 template <typename Impl>
-HandleFor<Impl, Struct> FactoryBase<Impl>::NewStruct(
-    InstanceType type, AllocationType allocation) {
-  Map map = Map::GetStructMap(read_only_roots(), type);
+Handle<Struct> FactoryBase<Impl>::NewStruct(InstanceType type,
+                                            AllocationType allocation) {
+  ReadOnlyRoots roots = read_only_roots();
+  Map map = Map::GetInstanceTypeMap(roots, type);
   int size = map.instance_size();
-  HeapObject result = AllocateRawWithImmortalMap(size, allocation, map);
-  HandleFor<Impl, Struct> str = handle(Struct::cast(result), isolate());
-  str->InitializeBody(size);
-  return str;
+  return handle(NewStructInternal(roots, map, size, allocation), isolate());
 }
 ```
 Every object that is stored on the v8 heap has a Map (`src/objects/map.h`) that
 describes the structure of the object being stored.
 ```c++
-class Map : public HeapObject {
+class Map : public TorqueGeneratedMap<Map, HeapObject> {
```

Comment:
Refer to https://github.com/v8/v8/blob/ee74e718339223345b42935e9f4cbc7be208e557/src/objects/map.h#L191